src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java

```diff
@@ -69,7 +69,7 @@ default boolean isCorrectToolForDrops(BlockState state, final ItemStack stack, B
             {
                 return stack.isCorrectToolForDrops(state);
             }
-            Block block = mtbe.getTextureData().getTexturedComponents().get(getMainComponent().getId());
+            Block block = mtbe.getTextureData().texturedComponents().get(getMainComponent().getId());
```

Comment:
let's not rename this getter. This just creates a huge dif

Reply:
Record issue, can name the record var as getter

Reply:
just convert it to a class then

Reply:
DataComps should be strictly immutable class = record, or act as such due to vanilla, it was class before

Reply:
It's fine to be a class, just make stuff final. Or do they enforce record?

Reply:
As class it must have full data equals() so just like I said "to act as record" but can be mutable

Reply:
Probably easier to leave it a class then with final and equal method. records are always a pain fi we need sth else later too.

Reply:
ye records are just syntax sugar, we can make an equivalent class with the same requirements

Reply:
I'll merge this and do a followup PR with the change, as that's faster

Reply:
> ye records are just syntax sugar, we can make an equivalent class with the same requirements, most IDE's even do the auto-conversion for you

That's imho void argument, literally the point of syntax sugar is to make code more understandable

Reply:
how is that a void argument when the point here is to keep the method name by using a record equivalent class?

Reply:
I also don't think records make code more understandable. They make it more compact.